src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/versionMatcher.hpp

```diff
@@ -0,0 +1,664 @@
+    {
+        std::deque<uint32_t> versionStrSplitA;
+        std::sregex_token_iterator itA(versionStrA.begin(), versionStrA.end(), parserVersionStrRegex, -1);
+        std::sregex_token_iterator itEndA;
+        while (itA != itEndA)
+        {
+            versionStrSplitA.push_back(static_cast<uint32_t>(std::stoul(itA->str())));
+            itA++;
+        }
+
+        std::deque<uint32_t> versionStrSplitB;
+        std::sregex_token_iterator itB(versionStrB.begin(), versionStrB.end(), parserVersionStrRegex, -1);
+        std::sregex_token_iterator itEndB;
+        while (itB != itEndB)
+        {
+            versionStrSplitB.push_back(static_cast<uint32_t>(std::stoul(itB->str())));
+            itB++;
+        }
+
+        while (versionStrSplitA.size() < versionStrSplitB.size())
+        {
+            versionStrSplitA.push_back(0);
+        }
+
+        while (versionStrSplitB.size() < versionStrSplitA.size())
+        {
+            versionStrSplitB.push_back(0);
+        }
+
+        for (size_t itemIdx; itemIdx < versionStrSplitA.size(); itemIdx++)
+        {
+            if (versionStrSplitA[itemIdx] < versionStrSplitB[itemIdx])
+            {
+                return -1;
+            }
+            if (versionStrSplitA[itemIdx] > versionStrSplitB[itemIdx])
+            {
+                return 1;
+            }
+        }
+
+        return 0;
+    }
+
+public:
+    VersionObjectPEP440(const std::string& version)
+    {
+        std::smatch parserMatches;
+        if ((std::regex_match(version, parserMatches, parserRegex) == false) || (parserMatches.size() != 7))
+        {
+            throw std::runtime_error {"Error creating " + std::string(typeid(*this).name())};
+        }
+
+        if (parserMatches.str(1).size())
+        {
+            epoch = static_cast<uint32_t>(std::stoul(parserMatches.str(1)));
+        }
+        else
+        {
+            epoch = 0;
+        }
+
+        versionStr = parserMatches.str(2);
+
+        if (parserMatches.str(3).size() && parserMatches.str(4).size())
+        {
+            preReleaseStr = parserMatches.str(3);
+            preReleaseNumber = static_cast<uint32_t>(std::stoul(parserMatches.str(4)));
+            hasPreRelease = true;
+        }
+        else
+        {
+            preReleaseNumber = 0;
+            hasPreRelease = false;
+        }
+
+        if (parserMatches.str(5).size())
+        {
+            postReleaseNumber = static_cast<uint32_t>(std::stoul(parserMatches.str(5)));
+            hasPostRelease = true;
+        }
+        else
+        {
+            postReleaseNumber = 0;
+            hasPostRelease = false;
+        }
+
+        if (parserMatches.str(6).size())
+        {
+            devReleaseNumber = static_cast<uint32_t>(std::stoul(parserMatches.str(6)));
+            hasDevRelease = true;
+        }
+        else
+        {
+            devReleaseNumber = 0;
+            hasDevRelease = false;
+        }
```

Comment:
It may be more readable. Something like this:
```
    epoch = parserMatches.str(1).empty() ? 0 : std::stoul(parserMatches.str(1));
    versionStr = parserMatches.str(2);

    preReleaseStr = parserMatches.str(3);
    preReleaseNumber = parserMatches.str(4).empty() ? 0 : std::stoul(parserMatches.str(4));
    hasPreRelease = !preReleaseStr.empty();

    postReleaseNumber = parserMatches.str(5).empty() ? 0 : std::stoul(parserMatches.str(5));
    hasPostRelease = postReleaseNumber > 0;

    devReleaseNumber = parserMatches.str(6).empty() ? 0 : std::stoul(parserMatches.str(6));
    hasDevRelease = devReleaseNumber > 0;
```

Reply:
done with some corrections, but keeping the concept